### PR TITLE
length of longest substring

### PR DESCRIPTION
**Pull request by mengfansheng-git**
length of longest substring

### CODE REVIEW

File: src/string/length_of_longest_substring.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+Example 3.
+Input: s = "pwwkew"
+Output: 3
+Explanation: Since the longest substring without repeated characters is "wke", its length is 3.
+Note that your answer must be the length of the substring; "pwke" is a subsequence, not a substring.
+*/
+
+pub fn length_of_longest_substring(s: String) -> i32 {
+    let s = s.as_bytes();
```
[the remark this answers is not in the document]
Comment (realstealthninja) [reply src/string/length_of_longest_substring.rs:21]: you'll need to change the rest of the code as well if you wish to stick to utf8

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Comment by realstealthninja** [comment]
you could also opt for the option to have results and check for if the index overflows and raise an error instead of causing a panic

**Comment by realstealthninja** [comment]
@vil02  what do you think? should this be just paniced once the function recieves a non ascii character or refactored to work with chars instead of bytes?

**Comment by vil02** [comment]
> @vil02 what do you think? should this be just paniced once the function recieves a non ascii character or refactored to work with chars instead of bytes?

Since the `String` is used, I would change this implementation to work with any chars.

**Comment by mengfansheng-git** [comment]
Do I need to remove the panic and change the logic of the whole function for the characters, does function name reviewers have a better suggestion?

**Pull request closed without merging**